src/Compiler/Checking/Expressions/CheckSequenceExpressions.fs

```diff
@@ -444,20 +444,12 @@ let TcSequenceExpressionEntry (cenv: TcFileState) env (overallTy: OverallTy) tpe
     match RewriteRangeExpr comp with
     | Some replacementExpr -> TcExpr cenv overallTy env tpenv replacementExpr
     | None ->
-
         let implicitYieldEnabled =
             cenv.g.langVersion.SupportsFeature LanguageFeature.ImplicitYield
 
         let validateObjectSequenceOrRecordExpression = not implicitYieldEnabled
 
         match comp with
-        | SynExpr.New _ ->
```

Comment:
To help reviewers: This specific change was removed in https://github.com/dotnet/fsharp/pull/17388/files but added back in previous refactoring PR by mistake.